src/adapters/suggest.js

```diff
@@ -1,258 +1,284 @@
-    }
-    return suggestList;
+  async componentDidMount() {
+    const favorites = await PoiStore.getAll();
+    this.setState({
+      favorites,
+    });
   }
 
-  setIdle(idle) {
-    this.searchInputDomHandler.readOnly = idle;
+  componentDidUpdate(prevProps) {
+    // if (this.props.inputValue !== prevProps.inputValue) {
+    //   this.source(this.props.inputValue);
+    // }
   }
 
-  async onSubmit() {
-    if (this.pending) {
-      if (this.bragiPromise) {
-        this.bragiPromise.abort();
-      }
-      const term = this.searchInputDomHandler.value;
-      this.preselect(term);
+  source = debounce(async term => {
+    let promise;
+
+    if (term === '') {
+      // Prerender Favorites on focus in empty field
+      // console.log('should Prerender Favorites');
+      // const favorites = await PoiStore.getAll();
+      // console.log('favorites', favorites);
+      this.setState({
+        pois: [],
+      });
     } else {
-      if (this.suggestList && this.suggestList.length > 0 &&
-          this.searchInputDomHandler.value &&
-          this.searchInputDomHandler.value.length > 0) {
-        this.onSelect(this.suggestList[0]);
-        this.searchInputDomHandler.blur();
-      }
-    }
-  }
-
-  destroy() {
-    this.autocomplete.destroy();
-  }
-
-  preRender(items) {
-    this.autocomplete.preRender(items);
-  }
-
-  prefixesRender() {
-    return this.prefixes.map(prefix => prefix.render());
-  }
-
-  remotesRender(pois) {
-    return pois.map(poi => this.renderItem(poi)).join('');
-  }
+      promise = () => new Promise(async (resolve, reject) => {
+        const focus = {};
+        const mapZoom = window.map && window.map.mb && window.map.mb.getZoom();
+        if (SUGGEST_USE_FOCUS && mapZoom >= SUGGEST_FOCUS_MIN_ZOOM) {
+          const center = window.map.mb.getCenter();
+          focus.lat = center.lat;
+          focus.lon = center.lng;
+          focus.zoom = mapZoom;
+        }
+        this.historyPromise = PoiStore.get(term);
+        this.bragiPromise = BragiPoi.get(term, focus);
+        this.categoryPromise = this.props.withCategories ?
+          CategoryService.getMatchingCategories(term) : null;
+
+        try {
+          const [bragiResponse, storeResponse, categoryResponse] = await Promise.all([
+            this.bragiPromise, this.historyPromise, this.categoryPromise,
+          ]);
+
+          if (!bragiResponse) {
+            return resolve(null);
+          }
+
+          this.suggestList = [];
+
+          if (categoryResponse) {
+            // this.suggestList = this.suggestList.concat(categoryResponse);
+            this.setState({
+              categories: categoryResponse,
+            });
+          }
+
+          this.bragiPromise = null;
+          this.suggestList = this.suggestList.concat(bragiResponse);
+          this.suggestList = this.suggestList.concat(storeResponse);
+
+          // console.log(this.suggestList);
+
+          resolve(this.suggestList);
```

Comment:
This whole promise could be simplified a lot so we see clearer :) And we don't need to store suggestList anymore.